Some 32-bit platforms need explicit linking to `libatomic` for 8-byte atomics support

Some 32-bit platforms (`ppc`, `arm`, likely `mips`) need explicit linking to `libatomic` here:
```
[100%] Linking CXX executable FuzzFailTest
cd /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_CLI11/CLI11/work/build/tests && /opt/local/bin/cmake -E cmake_link_script CMakeFiles/FuzzFailTest.dir/link.txt --verbose=ON
/opt/local/bin/g++-mp-13 -pipe -Os -DNDEBUG -I/opt/local/include -D_GLIBCXX_USE_CXX11_ABI=0 -mmacosx-version-min=10.6 -Wl,-search_paths_first -Wl,-headerpad_max_install_names -L/opt/local/lib -Wl,-headerpad_max_install_names CMakeFiles/FuzzFailTest.dir/FuzzFailTest.cpp.o CMakeFiles/FuzzFailTest.dir/__/fuzz/fuzzApp.cpp.o -o FuzzFailTest  -Wl,-rpath,/opt/local/lib /opt/local/lib/libCatch2Main.a /opt/local/lib/libCatch2.a 
Undefined symbols:
  "___atomic_store_8", referenced from:
      __ZNSt17_Function_handlerIFbRKSt6vectorISsSaISsEEEZN3CLI3App8add_flagISt6atomicIdELNS6_6detail7enablerE0EEEPNS6_6OptionESsRT_SsEUlS4_E_E9_M_invokeERKSt9_Any_dataS4_ in fuzzApp.cpp.o
      __ZNSt17_Function_handlerIFbRKSt6vectorISsSaISsEEEZN3CLI3App10add_optionISt6atomicIxESA_LNS6_6detail7enablerE0EEEPNS6_6OptionESsRT_SsEUlS4_E_E9_M_invokeERKSt9_Any_dataS4_ in fuzzApp.cpp.o
      __ZNSt17_Function_handlerIFbRKSt6vectorISsSaISsEEEZN3CLI3App10add_optionISt6atomicIyESA_LNS6_6detail7enablerE0EEEPNS6_6OptionESsRT_SsEUlS4_E_E9_M_invokeERKSt9_Any_dataS4_ in fuzzApp.cpp.o
  "___atomic_load_8", referenced from:
      __ZNSt17_Function_handlerIFSsvEZN3CLI3App10add_optionISt6atomicIxES5_LNS1_6detail7enablerE0EEEPNS1_6OptionESsRT_SsEUlvE_E9_M_invokeERKSt9_Any_data in fuzzApp.cpp.o
      __ZNSt17_Function_handlerIFSsvEZN3CLI3App10add_optionISt6atomicIyES5_LNS1_6detail7enablerE0EEEPNS1_6OptionESsRT_SsEUlvE_E9_M_invokeERKSt9_Any_data in fuzzApp.cpp.o
ld: symbol(s) not found
collect2: error: ld returned 1 exit status
make[2]: *** [tests/FuzzFailTest] Error 1
```

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -188,6 +188,18 @@ if(CMAKE_CXX_STANDARD GREATER 16)
     target_compile_definitions(FuzzFailTest_Single PUBLIC -DTEST_FILE_FOLDER="${TEST_FILE_FOLDER}")
     target_sources(FuzzFailTest_Single PUBLIC ${PROJECT_SOURCE_DIR}/fuzz/fuzzApp.cpp)
   endif()
+
+  # Some platforms need to link to atomic
+  file(WRITE "${CMAKE_BINARY_DIR}/test_atomic.cpp"
+       "#include <atomic>\n" "int main() { std::atomic<int64_t> i(0); i++; return 0; }\n")
+  try_compile(ATOMIC_BUILD_SUCCEEDED "${CMAKE_BINARY_DIR}" "${CMAKE_BINARY_DIR}/test_atomic.cpp")
+  if(NOT ATOMIC_BUILD_SUCCEEDED)
+    target_link_libraries(FuzzFailTest PRIVATE atomic)
+    if(CLI11_SINGLE_FILE AND CLI11_SINGLE_FILE_TESTS)
+      target_link_libraries(FuzzFailTestSingle PRIVATE atomic)
+    endif()
+  endif()
+  file(REMOVE ${CMAKE_BINARY_DIR}/test_atomic.cpp)
 endif()
 
 # Add -Wno-deprecated-declarations to DeprecatedTest

